pkg/translator/prometheus/normalize_name.go

```diff
@@ -192,6 +194,136 @@ func normalizeName(metric pmetric.Metric, namespace string) string {
 	return normalizedName
 }
 
+func normalizeSingleUnit(unit string, unitMapper func(unit string) string) string {
+	unit = strings.TrimSpace(unit)
+	if unit == "" || strings.ContainsAny(unit, "{}") {
+		return ""
+	}
+
+	return CleanUpString(unitMapper(unit))
+}
+
+func otlpUnitToProm(otlpUnit string) []string {
+	// Split unit at the '/' if any
+	unitTokens := strings.SplitN(otlpUnit, "/", 2)
+	if len(unitTokens) < 1 {
+		return nil
+	}
+
+	// Main unit
+	mainUnit := normalizeSingleUnit(unitTokens[0], unitMapGetOrDefault)
+	if len(unitTokens) == 1 {
+		if mainUnit == "" {
+			return nil
+		}
+
+		return []string{mainUnit}
+	}
+
+	secondaryUnit := normalizeSingleUnit(unitTokens[1], perUnitMapGetOrDefault)
+	if secondaryUnit != "" && mainUnit != "" {
+		return []string{mainUnit, "per", secondaryUnit}
+	}
+
+	// At this point, either mainUnit or secondaryUnit is empty, or both.
+	// Whichever is not empty gets returned, if none, return empty.
+	if mainUnit != "" {
+		return []string{mainUnit}
+	}
+
+	if secondaryUnit != "" {
+		return []string{"per", secondaryUnit}
+	}
+
+	return nil
+}
+
+type Normalizer struct {
+	registry *featuregate.Registry
+}
+
+func NewNormalizer(registry *featuregate.Registry) *Normalizer {
+	return &Normalizer{
+		registry: registry,
+	}
+}
+
+// TrimPromSuffixes trims type and unit prometheus suffixes from a metric name.
+// Following the [OpenTelemetry specs] for converting Prometheus Metric points to OTLP.
+//
+// [OpenTelemetry specs]: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/data-model.md#metric-metadata
+func (n *Normalizer) TrimPromSuffixes(promName string, metricType pmetric.MetricType, unit string) string {
+	if !n.registry.IsEnabled(normalizeNameGateID) {
+		return promName
+	}
+
+	nameTokens := strings.Split(promName, "_")
+	if len(nameTokens) == 1 {
+		return promName
+	}
+
+	nameTokens = removeTypeSuffixes(nameTokens, metricType)
+	nameTokens = removeUnitSuffixes(nameTokens, unit)
+
+	return strings.Join(nameTokens, "_")
+}
+
+func removeTypeSuffixes(tokens []string, metricType pmetric.MetricType) []string {
```

Comment:
Maybe add a note that only counters are expected to have suffixes present in the TYPE comment (whereas, histograms, for example, should not):

E.g.
```
# TYPE my_metric_total counter
my_metric_total{} 1
```
vs 
```
# TYPE my_metric histogram
my_metric_sum{} 1
my_metric_count{} 1
```

Reply:
That explains why they are the only type that has this logic applied.

Reply:
I thought that we'd to trim the `_total` from counters because those are not done naturally during the scraping/parsing, like histograms `_count` and `_sum` metric data points are parsed directly at:

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/prometheusreceiver/internal/metricfamily.go#L261-L268

In a sense, `_count`, `_sum` and `_bucket` (even `_created`?) represents fields of the underlying metric structure while `_total` is just an informative suffix for a counter value.

Also, from the [OpenMetrics specs](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#counter-1), I thought that even counters, shouldn't have the type suffix as part of the metric family name, as:
```
# TYPE foo counter
foo_total 17.0
```

ps: sorry for my lack of knowledge on the OpenMetrics terminology If I got some wrong.

Reply:
You are correct for OpenMetrics not having _total suffixes.  In prometheus, counters _are_ expected to have _total suffixes.  We already trim total suffixes from OM counters in the scrape logic, so this logic you are adding is just for the prometheus case.

Reply:
That seems like a bug...  Is there a different http_connected_total metric somewhere?  Prom shouldnt be able to find metadata for `http_connected_total`...

Reply:
Can you open an issue to track it?  Thanks a bunch for looking